### PR TITLE
Remove useMutableSource

### PR DESCRIPTION
**Pull request by noahlemen**
## Summary

This PR cleans up `useMutableSource`. This has been blocked by a remaining dependency internally at Meta, but that has now been deleted.

## How did you test this change?

```
yarn flow
yarn lint
yarn test --prod
```

### CODE REVIEW

File: packages/react-reconciler/src/ReactFiberLane.js [hunk 1]
```diff
@@ -299,8 +299,8 @@ export function getNextLanes(root: FiberRoot, wipLanes: Lanes): Lanes {
   // time it takes to show the final state, which is what they are actually
   // waiting for.
   //
-  // For those exceptions where entanglement is semantically important, like
-  // useMutableSource, we should ensure that there is no partial work at the
```
Comment (kassens) [inline packages/react-reconciler/src/ReactFiberLane.js:303]: @acdlite are there remaining cases of this? Maybe the replacement, `useSyncExternalStore` or can this block be removed?

File: packages/shared/ReactVersion.js [hunk 2]
```diff
@@ -1,16 +1 @@
...
-// TODO: this is special because it gets imported during build.
-//
-// TODO: 18.0.0 has not been released to NPM;
-// It exists as a placeholder so that DevTools can support work tag changes between releases.
-// When we next publish a release, update the matching TODO in backend/renderer.js
-// TODO: This module is used both by the release scripts and to expose a version
-// at runtime. We should instead inject the version number as part of the build
-// process, and use the ReactVersions.js module as the single source of truth.
-export default '18.2.0';
+export default '18.3.0-PLACEHOLDER';
```
Comment (kassens) [inline packages/shared/ReactVersion.js:1]: revert, this is an annoying artifact of the build process that it overrides this file…

File: packages/react-reconciler/src/ReactFiberLane.js [hunk 3]
```diff
@@ -622,10 +622,6 @@ export function markRootPinged(root: FiberRoot, pingedLanes: Lanes) {
   root.pingedLanes |= root.suspendedLanes & pingedLanes;
 }
 
-export function markRootMutableRead(root: FiberRoot, updateLane: Lane) {
-  root.mutableReadLanes |= updateLane & root.pendingLanes;
```
Comment (kassens) [inline packages/react-reconciler/src/ReactFiberLane.js:626]: Can you grep for `mutableReadLanes`, I think it can be removed from the fiber too.

**Review by sebmarkbage — APPROVED** [review]

**Comment by merouanezouaid** [comment]
RIP ⚰️

**Pull request merged**